src/compiler/checker.ts

```diff
@@ -25891,6 +25891,10 @@ export function createTypeChecker(host: TypeCheckerHost): TypeChecker {
                 if (!nonConstructorTypeInUnion) return type;
             }
 
+            if (assumeTrue && declaredType === unknownType && isEmptyAnonymousObjectType(type)) {
+                return targetType;
+            }
+

```

Comment:
The problem with conditioning this on the declared type is that it doesn't work when the empty anonymous object type originates in some other type. For example:

```ts
function f<T>(x: T & {}) {
    return x instanceof Object && 'a' in x;  // Error, but shouldn't be
}
```